core/common/src/main/java/alluxio/util/EnvironmentUtils.java

```diff
@@ -45,5 +59,197 @@ public static boolean isKubernetes() {
+  private static boolean isEC2WithProductUUID() {
+    try {
+      return fileExistAndStartWithIdentifier("/sys/devices/virtual/dmi/id/product_uuid", "EC2");
+    } catch (Throwable t) {
+      // Exceptions are expected if this instance is not EC2 instance
+      // or this check is not valid
+      return false;
+    }
+  }
+
+  /**
+   * Check if this instance is an EC2 instance with instance identity.
+   * Note that other cloud providers make this instance metadata URL available
+   * and may thus cause false positive.
+   *
+   * @return true if jvm runs in EC2 instance with instance identity, false otherwise
+   */
+  private static boolean isEC2WithInstanceIdentity() {
+    try {
+      EC2MetadataUtils.getInstanceInfo();
+      return true;
+    } catch (Throwable t) {
+      // Exceptions are expected if this instance is not EC2 instance
+      // or this check is not valid
+      return false;
+    }
+  }
+
+  /**
+   * @return true, if running on GCE with google metadata available
+   */
+  private static boolean isGCEWithMetadata() {
+    try {
+      String url = "http://metadata.google.internal/computeMetadata/v1/instance/zone";
+      HttpGet post = new HttpGet(url);
+      post.setHeader("Metadata-Flavor", "Google");
+      HttpClient client = HttpClientBuilder.create()
+          .setDefaultRequestConfig(
+              RequestConfig.custom()
+                  .setConnectionRequestTimeout(1000)
+                  .setConnectTimeout(1000)
+                  .setSocketTimeout(1000)
+                  .build())
+          .build();
+      HttpResponse response = client.execute(post);
+
+      int responseCode = response.getStatusLine().getStatusCode();
+      if (responseCode != HttpURLConnection.HTTP_OK) {
+        return false;
+      }
+      String zone = EntityUtils.toString(response.getEntity(), "UTF-8");
+      return !zone.isEmpty();
+    } catch (Throwable t) {
+      // Exceptions are expected if this instance is not GCE instance
+      // or this GCE does not allow fetching metadata
+      return false;
+    }
+  }
+
+  /**
+   * @return true, if running on GCE with Google bios vendor
+   */
+  private static boolean isGCEWithBiosVendor() {
+    try {
+      Process process = Runtime.getRuntime().exec("sudo dmidecode -s bios-vendor");
+      try (Reader reader = new InputStreamReader(process.getInputStream())) {
+        String output = CharStreams.toString(reader);
+        return output.contains("Google");
+      }
+    } catch (Throwable t) {
+      // Exceptions are expected if this instance is not
+      // running on Google bios vendor
+      return false;
+    }
+  }
+
+  /**
+   * Checks if the target file exists and starts with the identifier string.
+   *
+   * @param filePath the file to check
+   * @param identifier the string identifier to check
+   * @return true if file exists and starts with the identifier
+   */
+  private static boolean fileExistAndStartWithIdentifier(String filePath,
```

Comment:
nit: rename to `ec2UUIDFileExistsWithID`?

Reply:
Done.